clang/lib/CConv/ProgramInfo.cpp

```diff
@@ -413,18 +411,18 @@ bool ProgramInfo::link() {
   for (const auto &U : ExternFunctions) {
     // If we've seen this symbol, but never seen a body for it, constrain
     // everything about it.
-    if (!U.second && !isExternOkay(U.first)) {
+    std::string FuncName = U.first;
+    if (!U.second && !isExternOkay(FuncName)) {
       // Some global symbols we don't need to constrain to wild, like 
       // malloc and free. Check those here and skip if we find them. 
-      std::string FuncName = U.first;
-      auto FuncDeclFVIterator = ExternalFunctionFVCons.find(FuncName);
-      assert(FuncDeclFVIterator != ExternalFunctionFVCons.end());
-      const std::set<FVConstraint *> &Gs = (*FuncDeclFVIterator).second;
+      const std::set<FVConstraint *>
+          *Gs = getExtFuncDefnConstraintSet(FuncName);
+      assert("Function constraints could not be found!" && Gs != nullptr);
 
       // If there was a checked type on a variable in the input program, it
       // should stay that way. Otherwise, we shouldn't be adding a checked type
       // to an extern function.
-      for (auto *const G : Gs) {
+      for (auto *const G : *Gs) {
```

Comment:
I'm surprised about this; why was it working before?

Reply:
I changed this to use the existing `getExtFuncDefnConstraintSet` method which returns a pointer, so the extra dereference had to be introduced.